File: src/data/types/crewCandidate.ts
import type { TierColor } from "./tierColor.ts";
import type { Profession } from "./profession.ts";

// Necessary correction to the Phase 4 amendment, discovered while
// implementing Agent 16: PlanetCrewPool.availableHires was typed as
// CrewMember[], but an unhired pool candidate has no real
// hiredByPlayerId/hiredAt/wageAmount/lastPaidAt yet -- those fields are
// only meaningful once hireCrew() actually hires someone. CrewCandidate
// is the "browsable, not-yet-hired" shape; hireCrew() is what turns one
// into a real CrewMember (assigning the hire-specific fields at that
// moment). This is a field-type correction, not a pure addition, unlike
// this project's usual necessary-completion pattern -- documented
// prominently rather than silently forcing fake values into CrewMember's
// required fields to preserve the letter of the original pseudocode.
export interface CrewCandidate {
  id: string;
  tier: TierColor;
  profession: Profession | null;
}

File: unity/ProfitableCore/Schema/CrewCandidate.cs
namespace Profitable.Core.Schema;

// Ports src/data/types/crewCandidate.ts. The "browsable, not-yet-hired"
// shape -- HireCrew is what turns one into a real CrewMember (assigning
// the hire-specific fields at that moment). Profession is null for tiers
// 1-5 (Grey-Purple, general/unspecialized); rolled from the tier 6-7
// taxonomy only for Orange/Gold candidates.
public class CrewCandidate
{
    public string Id { get; set; } = string.Empty;
    public TierColor Tier { get; set; }
    public string? Profession { get; set; }
}
